docs/excel/excel-add-ins-pivottables.md

```diff
@@ -175,6 +176,67 @@ The following code samples changes the aggregation to be averages of the data.
     });
 ```
 
+## Change calculations with a ShowAsRule
+
+PivotTables, by default, aggregate the data of their row and column hierarchies independently. A `ShowAsRule` changes the data hierarchy to output values based on other items in the PivotTable.
+
+The `ShowAsRule` object has three properties:
+-	`calculation`: The type of relative calculation to apply to the data hierarchy (the default is `none`).
+-	`baseField`: The field within the hierarchy containing the base data before the calculation is applied. The `PivotField` usually has the same name as its parent hierarchy.
+-	`baseItem`: The individual item compared against the values of the base fields based on the calculation type. Not all calculations require this field.
+
+The following example sets the calculation on the **Sum of Crates Sold at Farm** data hierarchy to be a percentage of the column total. 
+We still want the granularity to extend to the fruit type level, so we’ll use the **Type** row hierarchy and its underlying field. 
+The example also has **Farm** as the first row hierarchy, so the farm total entries display the percentage each farm is responsible for producing as well.
+
+![A PivotTable showing the percentages of fruit sales relative to the grand total for both individual farms and individual fruit types within each farm.](../images/excel-pivots-showas-percentage.png)
+
+``` TypeScript
+    await Excel.run(async (context) => {
+        const pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+        const farmDataHierarchy = pivotTable.dataHierarchies.getItem("Sum of Crates Sold at Farm");
+
+        farmDataHierarchy.load("showAs");
+        await context.sync();
+
+        // show the crates of each fruit type sold at the farm as a percentage of the column's total
+        let farmShowAs = farmDataHierarchy.showAs;
+        farmShowAs.calculation = Excel.ShowAsCalculation.percentOfColumnTotal;
+        farmShowAs.baseField = pivotTable.rowHierarchies.getItem("Type").fields.getItem("Type");
+        farmDataHierarchy.showAs = farmShowAs; 
+        farmDataHierarchy.name = "Percentage of Total Farm Sales";
+
+        await context.sync();
+    });
+```
+
+The previous example set the calculation to the column, relative to an individual row hierarchy. When the calculation relates to an individual item, use the `baseItem` property. 
+
+The following example shows the `differenceFrom` calculation. It displays the difference of the farm crate sales data hierarchy entries relative to those of “A Farms”. 
+The `baseField` is **Farm**, so we see the differences between the other farms, as well as breakdowns for each type of like fruit (**Type** is also a row hierarchy in this example).
+
+``` TypeScript
+    await Excel.run(async (context) => {
```

Comment:
Code block should be left-aligned within the TypeScript box -- i.e., move this entire code block to the left by 4 spaces so that it's left-aligned.